### PR TITLE
Enhance createToken functionality with authentication checks using Privy

### PR DESCRIPTION
**Pull request by Nathaniyell**
1. Added new response statuses: MISSING_AUTH, MISSING_WALLET, and ERROR to CreateTokenResponse enum.
2. Implemented authentication checks using usePrivy to ensure the user is connected before proceeding with token creation.
3. Improved error handling for wallet address retrieval and transaction preparation, providing clearer feedback to the user.

### CODE REVIEW

File: lib/tools/createToken.ts [hunk 1]
```diff
@@ -20,43 +19,46 @@ const createToken = (question: string) => tool({
     mimeType: z.string().optional().describe("The type of media."),
   }),
   execute: async ({ image, title, collectionAddress, mimeType }) => {
-    if (!image || !mimeType) {
+    const validationErrors = [
+      { condition: !auth.authenticated, status: CreateTokenResponse.MISSING_AUTH, message: "Please connect your wallet first." },
+      { condition: !auth.user?.wallet?.address, status: CreateTokenResponse.MISSING_WALLET, message: "No wallet address found." },
```
Comment (techeng322) [inline lib/tools/createToken.ts]: Question: ❓ 
Can we pass only address from privy to check MISSING_AUTH & MISSING_WALLET?
Tool call param: image, title, mimeType, collectionAddress, walletAddress
Status: MISSING_CONNECTED_WALLET

  `const { login, user } = usePrivy();
  const address = user?.wallet?.address as Address;`

File: lib/getContext.ts [hunk 2]
```diff
@@ -2,10 +2,14 @@ const getContext = async (address: string) => {
     const BASE_URL = 'https://api.myco.wtf';
 
     const [tokensResponse, scoreResponse] = await Promise.all([
-        fetch(`${BASE_URL}/api/zora/tokens?creatorAddress=${address}`),
-        fetch(`${BASE_URL}/api/zora/score?address=${address}`)
+        fetch(`${BASE_URL}/api/zora/tokens?privyWalletAddress=${address}`),
+        fetch(`${BASE_URL}/api/zora/score?privyWalletAddress=${address}`)
```
Comment (sweetmantech) [inline lib/getContext.ts]: Why are you changing these params?
- Did you update each respective API to accept a different parameter name `privyWalletAddress`?

File: lib/getContext.ts [hunk 3]
```diff
@@ -2,10 +2,14 @@ const getContext = async (address: string) => {
...
     ]);
 
+    if (!tokensResponse.ok || !scoreResponse.ok) {
+        throw new Error('Failed to fetch data');
+    }
```
Comment (sweetmantech) [inline lib/getContext.ts]: I do not think you need to be making any changes to this file.

File: lib/server/chat-messages.service.ts [hunk 4]
```diff
@@ -11,11 +11,18 @@ export function createChatMessagesService() {
...
+  private address!: Address;
+
   constructor() {}
 
-  async getChatSettings(chatReferenceId: string, address: Address, question: string) {
-    const context = await this.fetchRelevantContext(address);
-    const tools = this.fetchRelevantTools(question);
+  async getChatSettings(
+    chatReferenceId: string, 
+    privyAddress: Address,
```
Comment (sweetmantech) [inline lib/server/chat-messages.service.ts]: why do you need to refactor this param to `privyAddress`?
- so far your changes seem to be non-essential refactors.
- Please focus on `simple` and `small` changes to focus on the requirements in the ticket. 🎟️

File: lib/server/chat-messages.service.ts [hunk 5]
```diff
@@ -11,11 +11,18 @@ export function createChatMessagesService() {
...
+    messageContent: string
```
Comment (sweetmantech) [inline lib/server/chat-messages.service.ts]: same feedback here. Why the refactor?

File: lib/server/chat-messages.service.ts [hunk 6]
```diff
@@ -32,7 +39,7 @@ Please use this information to provide accurate and relevant responses and don't
     };
   }
 
-  private async fetchRelevantContext(address: Address = "0xcfBf34d385EA2d5Eb947063b67eA226dcDA3DC38"): Promise<string> {
+  private async fetchRelevantContext(address: Address): Promise<string> {
```
Comment (sweetmantech) [inline lib/server/chat-messages.service.ts]: why did you remove the `fallback` value?

File: lib/toolResponse.types.ts [hunk 7]
```diff
@@ -3,4 +3,7 @@ export enum CreateTokenResponse {
   MISSING_TITLE = "MISSING_TITLE",
   MISSING_COLLECTION = "MISSING_COLLECTION",
   SIGN_TRANSACTION = "SIGN_TRANSACTION",
+  MISSING_AUTH = "MISSING_AUTH",
+  MISSING_WALLET = "MISSING_WALLET",
```
Comment (sweetmantech) [inline lib/toolResponse.types.ts]: What's the difference between `MISSING_AUTH` & `MISSING_WALLET`?
- When will they be different?
